src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
             return NO_TYPE_FILTER;
         }
 
-        var valLower = query.query.toLowerCase(),
-            val = valLower,
-            typeFilter = itemTypeFromName(query.type),
-            results = {}, results_in_args = {}, results_returned = {},
-            split = valLower.split("::");
+        userQuery = userQuery.trim();
+        var query = {
+            original: userQuery,
+            userQuery: userQuery.toLowerCase(),
+            length: userQuery.length,
+            pos: 0,
+            typeFilter: null,
+            elems: [],
+            elemName: null,
+            args: [],
+            returned: [],
+            // Total number of elements (includes generics).
+            totalElems: 0,
+            // Total number of "top" elements (does not include generics).
+            foundElems: 0,
+            // This field is used to check if it's needed to re-run a search or not.
+            id: "",
+            // This field is used in `sortResults`.
+            nameSplit: null,
```

Comment:
This seems to be a duplicate of `elems[0].fullPath`. And it seems like unnecessary coupling for `sortResults` to depend on this very specific value set by `parseQuery`. I think we should get rid of this field and have `sortResults` use `elems[0].fullPath` (or just `path` if you follow the simplification proposal I made for `elems`).

Reply:
I really wonder why I kept it there... Removed
 it.